src/content/en/2021/caching.md

```diff
@@ -17,4 +19,328 @@ featured_stat_3: TODO
+Here is an example of a valid Cache-Control header with multiple directives:
+
+
+```
+Cache-Control: public, max-age=86400, must-revalidate
+```
+
+Usage of this header has increased or remained consistent across web pages. 74% of Mobile and 75% of Desktop pages use Cache-Control.
+
+When using the Cache-Control header, usage increased by 1% for Desktop and remained the same for Mobile. However, usage of the Last-Modified header decreased by 1% across both interfaces.
+
+{{ figure_markup(
+  image="cache-control-adoption.png",
+  caption="The percent of requests that use the Cache-Control response header.",
+  description="TODO",
+  chart_url="https://docs.google.com/spreadsheets/d/e/2PACX-1vSGgVDZ9RkFQLmk5C3_siIcH-8macUEZMobcC0o1z8frYj8NOkI_C2s_yE5ppMdxDAD5INjNsCBa3h1/pubchart?oid=885292598&format=interactive",
+  sheets_gid="2102749619",
+  sql_file="header_trends.sql"
+) }}
+
+Cache Control directives, 61% of mobile requests include a Cache-Control response header with a `max-age` directive.
+
+The most misused Cache Control directive is `set-cookie`, used for 0.07% of total directives for Desktop and 0.08% for Mobile.
+
+Fun fact the the largest max-age value is 1625540701883604800000 seconds or 51545557517871.8 years (51 trillion).
+
+{{ figure_markup(
+  caption="TODO",
+  content="51 trillion years",
+  classes="big-number",
+  sheets_gid="529870849",
+  sql_file="todo.sql"
+) }}
+
+Talk about `immutable` directive, 12% of desktop requests containing Cache-Control: immutable are on the "static.parastorage.com" host.
+
+Talk about invalid date, last modified and expires, (0.11% of dates in desktop requests are invalid).
+
+Desktop - Valid date strings:
+99.18% using date, 72.57% using last-modified, 55.48% using expires
+
+Mobile - Valid date strings:
+99.07% using date, 70.55% using last-modified, 55.57% using expires
+
+Although 99% of both Desktop and Mobile pages use valid date strings, only 55% of Desktop and 56% of Mobile pages use 'expires'. Last-modified is used at a higher percentage, at 73% for Desktop and 71% for Mobile. While the amount of invalid 'date' and 'last-modified' directives is less than 1%, 2.75% of Mobile and 3.20% of Desktop 'expires' are being used incorrectly.
+
+What is or would be the impact from using expires more effectively?
+
+Could we possibly see big wins from getting more consistent here?
+
+### Conditional requests and revalidation
+
+### `ETag`
+
+<a hreflang="en" href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag">`ETag`</a> helps ....
+
+```
+ETag: "34a64df551429fcc55e4d42a148795d9f25f89d8"
+```
+
+### `Last-Modified`
+
+<a hreflang="en" href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Last-Modified">`Last-Modified`</a> helps ....
+
+```
+Last-Modified: Mon, 29 Nov 2021 07:28:00 GMT
+```
+
+### Vary
+
+Talk about Vary adoption, 42% of requests on mobile set Vary: accept-encoding.
+
```

Comment:
In addition, 83% of mobile Vary: accept-encoding requests also use cache-control. This is another case of the more you know about caching, the more consistently you can make use of all the resources available to you.